### PR TITLE
Some improves for layers editing in editor

### PR DESCRIPTION
**Pull request by BannZay**
Some fixes for editor while editing  switch layer
-removed unused entities
-fixed (added) rotation for switch indicators (number/delay) ( tiles has only rotation possibility before)
- added tiles rotation possibility for switch layer (Its useless but not less useless as for almost all game/front tiles)

### CODE REVIEW

**Comment by BannZay** [comment]
Here is styll very weird switch editing. 
First problem - When you grab some list of switch entities, will be founded the last valid entitie and their number and delay goes to switch-button values.
Tiny bug - switch-button values ovverride delay/number values only if it was changed to another one since grabbing. So if you will back to value which use on grabbing => it`ll have no override effect anymore.



I`m thinking about just removing overriding each entitie in brush by just changing values. But add (not sure which way. Probly binded button) another way to do it safety and easy to understand way.

**Comment by timakro** [comment]
a7c4121 is awesome!

File: src/game/mapitems.cpp [hunk 1]
```diff
@@ -37,6 +37,27 @@ bool IsValidFrontTile(int Index)
...
+		|| Index == TILE_HIT_START
+		|| Index == TILE_HIT_END
+		|| Index == TILE_SWITCHTIMEDOPEN
+		|| Index == TILE_SWITCHTIMEDCLOSE
+		|| Index == TILE_SWITCHOPEN
+		|| Index == TILE_SWITCHCLOSE
+		|| Index == TILE_PENALTY
+		|| Index == TILE_BONUS
+		//Switch layer do not contain spawns and flags (which acceptedeable in IsValidEntity() )
+		|| (Index >= ENTITY_ARMOR_1 + ENTITY_OFFSET && IsValidEntity(Index))
```
Comment (def-) [inline src/game/mapitems.cpp]: Just `IsValidEntity(Index)` works in this line as `IsValidEntity` already checks whether it's in the right range.

File: src/game/editor/layer_tiles.cpp [hunk 2]
```diff
@@ -1380,6 +1380,7 @@ CLayerSwitch::~CLayerSwitch()
 	delete[] m_pSwitchTile;
 }
 
+
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: Don't add random empty lines.

File: src/game/editor/editor.h [hunk 3]
```diff
@@ -1109,6 +1109,8 @@ class CLayerSwitch : public CLayerTiles
 	virtual void Shift(int Direction);
 	virtual void BrushDraw(CLayer *pBrush, float wx, float wy);
 	virtual void FillSelection(bool Empty, CLayer *pBrush, CUIRect Rect);
+	virtual void BrushRotate(float Amount);
+
```
Comment (def-) [inline src/game/editor/editor.h]: Another empty line

File: src/game/editor/layer_tiles.cpp [hunk 4]
```diff
@@ -1380,6 +1380,61 @@ CLayerSwitch::~CLayerSwitch()
...
+	int Rotation = (round_to_int(360.0f*Amount / (pi * 2)) / 90) % 4;	// 0=0°, 1=90°, 2=180°, 3=270°
+	if (Rotation < 0)
+		Rotation += 4;
+
+	if (Rotation == 1 || Rotation == 3)
+	{
+		// 90° rotation
+		CSwitchTile *pTempData = new CSwitchTile[m_Width*m_Height];
+		mem_copy(pTempData, m_pSwitchTile, m_Width*m_Height * sizeof(CSwitchTile));
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: What about m_pTiles? It's mem_copied in the other BrushRotate implementations.

File: src/game/editor/layer_tiles.cpp [hunk 5]
```diff
@@ -1380,6 +1380,61 @@ CLayerSwitch::~CLayerSwitch()
...
+			}
+
+		int Temp = m_Width;
+		m_Width = m_Height;
+		m_Height = Temp;
+		delete[] pTempData;
+	}
+
+
+
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: Spaces here and below

File: src/game/editor/layer_tiles.cpp [hunk 6]
```diff
@@ -1380,6 +1380,61 @@ CLayerSwitch::~CLayerSwitch()
...
+	if (Rotation == 2 || Rotation == 3)
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: These should be implemented as BrushFlipX() and BrushFlipY()

**Comment by BannZay** [comment]
I`m still working on it

**Comment by BannZay** [comment]
Have no time to finish it now. Closed for now

**Pull request closed without merging**